qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -160,32 +161,47 @@ def _component_data(self, composite_data: List[Dict]) -> List[List[Dict]]:
+    def _initialize_child_data(self, experiment_data: ExperimentData):
+        """Initialize component experiment data objects as child data"""
         component_index = experiment_data.metadata.get("component_child_index", [])
-        if not component_index:
-            experiment = experiment_data.experiment
-            if experiment is None:
-                raise AnalysisError(
-                    "Cannot run composite analysis on an experiment data without either "
-                    "a composite experiment, or composite experiment metadata."
-                )
-            # If the experiment Construct component data and update indices
-            start_index = len(experiment_data.child_data())
-            component_index = []
-            for i, sub_exp in enumerate(experiment.component_experiment()):
-                sub_data = sub_exp._initialize_experiment_data()
-                experiment_data.add_child_data(sub_data)
-                component_index.append(start_index + i)
-            experiment_data.metadata["component_child_index"] = component_index
-
-        # Child components exist so we can get their ID for accessing them
-        child_ids = experiment_data._child_data.keys()
-        component_ids = [child_ids[idx] for idx in component_index]
-        return component_ids
+        if component_index:
+            # Child components are already initialized
+            return
+
+        # Initialize the component experiment data containers and add them
+        # as child data to the current experiment data
+        child_components = self._initialize_component_data(experiment_data)
+        start_index = len(experiment_data.child_data())
+        for i, subdata in enumerate(child_components):
+            experiment_data.add_child_data(subdata)
+            component_index.append(start_index + i)
+
+        # Store the indices of the added child data in metadata
+        experiment_data.metadata["component_child_index"] = component_index
+
+    def _initialize_component_data(self, experiment_data: ExperimentData) -> List[ExperimentData]:
+        """Initialize component experiment data objects.
+
+        These contain the component metadata, and copy the tags, share level,
+        and auto save attributes of the main data.
+        """
+        # Extract component experiment types and metadata so they can be
+        # added to the component experiment data containers
+        metadata = experiment_data.metadata
+        num_components = len(self._analyses)
+        experiment_types = metadata.get("component_types", [None] * num_components)
+        component_metadata = metadata.get("component_metadata", [{}] * num_components)
+
+        # Create component experiments and copy backend, tags, share level
+        # and auto save from the parent experiment data
+        component_expdata = []
+        for i, _ in enumerate(self._analyses):
+            subdata = ExperimentData(backend=experiment_data.backend)
```

Comment:
How does this work if composite analysis is nested? Probably we need instance check of analysis and do recursive call of initialize method when composite analysis subclass is found.

Reply:
Nesting is fine. The point of doing it this way is that the analysis of each component is independent, so when the component analysis is run on the sub-experiment, if it is itself a composite experiment it will then go through this same initialization and for that component and its children.